odoo/addons/base/tests/test_ir_mail_server_smtpd.py

```diff
@@ -0,0 +1,460 @@
+        Certificate(key=certdir / 'client.key.pem', cert=certdir / 'client.cert.pem'),
+        Certificate(key=certdir / 'server.key.pem', cert=certdir / 'server.cert.pem'),
+        Certificate(key=certdir / 'self_signed.key.pem', cert=certdir / 'self_signed.cert.pem'),
+    )
+
+
+# skip when optional dependencies are not found
+@unittest.skipUnless(aiosmtpd, "aiosmtpd couldn't be imported")
+@unittest.skipUnless(_openssl, "openssl not found in path")
+# fail fast for timeout errors
+@patch('odoo.addons.base.models.ir_mail_server.SMTP_TIMEOUT', .1)
+# prevent the CLI from interfering with the tests
+@patch.dict('odoo.tools.config.options', {'smtp_server': ''})
+class TestIrMailServerSMTPD(TransactionCaseWithUserDemo):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # aiosmtpd emits deprecation warnings because it uses its own
+        # deprecated features, mute those logs.
+        # https://github.com/aio-libs/aiosmtpd/issues/347
+        class Session(aiosmtpd.smtp.Session):
+            @property
+            def login_data(self):
+                return self._login_data
+            @login_data.setter
+            def login_data(self, value):
+                self._login_data = value
+        patcher = patch('aiosmtpd.smtp.Session', Session)
+        patcher.start()
+        cls.addClassCleanup(patcher.stop)
+
+        # aiosmtpd emits warnings for some unusual configuration, like
+        # requiring AUTH on a clear-text transport. Mute those logs
+        # since we also test those unusual configurations.
+        warnings.filterwarnings(
+            'ignore',
+            "Requiring AUTH while not requiring TLS can lead to security vulnerabilities!",
+            category=UserWarning
+        )
+        class CustomFilter(logging.Filter):
+            def filter(self, record):
+                if record.msg == "auth_required == True but auth_require_tls == False":
+                    return False
+                if record.msg == "tls_context.verify_mode not in {CERT_NONE, CERT_OPTIONAL}; this might cause client connection problems":
+                    return False
+                return True
+        logging.getLogger('mail.log').addFilter(CustomFilter())
+
+        # decrease aiosmtpd verbosity, odoo INFO = aiosmtpd WARNING
+        logging.getLogger('mail.log').setLevel(_logger.getEffectiveLevel() + 10)
+
+        # create a few certificates for ssl/tls+starttls and mock the
+        # various smtplib classes into trusting the created root CA.
+        _logger.info("Using openssl to generate fake certificates, show "
+                     "subprocess output with: --log-handler %s:DEBUG", __name__)
+        tempdir = tempfile.TemporaryDirectory(prefix='odoo-test-smtpd-')
+        cls.addClassCleanup(tempdir.cleanup)
+        cls.ssl_ca, cls.ssl_client, cls.ssl_server, cls.ssl_self_signed = \
+            _generate_certificates(Path(tempdir.name))
```

Comment:
```suggestion
            _generate_certificates(tempdir)
```
and use `Path(tempdir.name)` inside the `_generate_certificates` method? (To avoid passing a simple string as the path to execute a script)